lib/iris/experimental/um.py

```diff
@@ -296,6 +316,64 @@ def set_data(self, data):
         """
         self._data_provider = data
 
+    def _raw_data_is_useable_with_lbpack(self, required_lbpack):
+        """
+        Return whether the field's raw payload can be reused unmodified,
+        for the specified output packing format.
+
+        """
+        if isinstance(self._data_provider, np.ndarray):
+            # The original data payload has been replaced by plain array data.
+            return False
+
+        existing_lbpack = self._data_provider.lookup_entry.lbpack
+        new_lbpack_upper, new_lbpack = lbpack_digits(required_lbpack)
+        old_lbpack_upper, old_lbpack = lbpack_digits(existing_lbpack)
+        # We can treat data word types 0, 2 and 3 as the same.
+        data_types_equivalent = (new_lbpack.n4 == old_lbpack.n4 or
+                                 ((new_lbpack.n4 in (0, 2, 3)) and
+                                  (old_lbpack.n4 in (0, 2, 3))))
+        # The packing words are compatible if nothing else is different.
+        compatible = (data_types_equivalent and
+                      new_lbpack_upper == old_lbpack_upper and
+                      ((new_lbpack.n3, new_lbpack.n2, new_lbpack.n1) ==
+                       (old_lbpack.n3, old_lbpack.n2, old_lbpack.n1)))
+        return compatible
+
+    def _check_valid_data_encoding(self, required_lbpack):
+        """
+        Check if we can save this field's data to a specified packing format.
+
+        If not, raise an explanatory exception.
+
+        """
+        new_lbpack_upper, new_lbpack = lbpack_digits(required_lbpack)
+
+        # Any data can be output as unpacked.
+        required_unpacked = (new_lbpack_upper == 0 and
+                             new_lbpack.n4 in (0, 2, 3) and
+                             new_lbpack.n3 == 0 and
+                             new_lbpack.n2 == 0 and
+                             new_lbpack.n1 == 0)
+        if required_unpacked:
+            return
+
+        # 'Else' the target is a packed format...
+        # Unpacked data can not currently be re-coded in any packed form.
+        if isinstance(self._data_provider, np.ndarray):
+            # An assigned array has replaced the original deferred data
+            # reference :  We can only cope with unpacked formats.
+            msg = ('Cannot save array data to format lbpack={} : '
+                   'requires unsupported encoding method')
+            raise ValueError(msg.format(new_lbpack))
+
+        # Packed data can be copied unchanged, to a compatible format.
+        if not self._raw_data_is_useable_with_lbpack(required_lbpack):
```

Comment:
It is not the same, the function does more than that :
 * disallow non-packed data
 * allows aliases of the word type ('native' / 'cray' / 'ieee')